### ISSUE TITLE
[BUG] `DIRECTORY.md` workflow not running due to protected branch

### ISSUE THREAD

**Issue by Panquesito7** [TheAlgorithms/JavaScript#1219]
**Description**

The [`DIRECTORY.md`](https://github.com/TheAlgorithms/JavaScript/blob/master/.github/workflows/UpdateDirectory.yml) workflow isn't committing/pushing any needed changes due to the `master` branch being protected.
Currently, there's no way to make GitHub Actions have access to a protected branch.

Our current options are:

- Disable branch protection (not recommended).
- Commit as someone with access to making direct commits to the branch.
- Wait until GitHub provides a solution to this.

**Comment by raklaptudirm** [comment]
Directory.md is currently changed in the prs themselves which I think is good enough.

**Comment by Panquesito7** [comment]
> Directory.md is currently changed in the prs themselves which I think is good enough.

Yeah. Although, I still think this should be kept open once GitHub actually fixes this.

**Comment by raklaptudirm** [comment]
The ideal scenario would be for the workflow to run only in the master.

**Comment by Panquesito7** [comment]
> The ideal scenario would be for the workflow to run only in the master.

That currently won't work as the `master` branch is protected.

**Comment by isaacudofia** [comment]
Maintainer / Owner access is never able to force push again for default & protected branch,

**Comment by rickydoo680** [comment]
> **Description**
> 
> The [`DIRECTORY.md`](https://github.com/TheAlgorithms/JavaScript/blob/master/.github/workflows/UpdateDirectory.yml) workflow isn't committing/pushing any needed changes due to the `master` branch being protected.
> Currently, there's no way to make GitHub Actions have access to a protected branch.
> 
> Our current options are:
> 
> - Disable branch protection (not recommended).
> - Commit as someone with access to making direct commits to the branch.
> - Wait until GitHub provides a solution to this.
> 
>

**Comment by Panquesito7** [comment]
@raklaptudirm, should I change the workflow to create a PR to the repository, just like in other repositories such as C/C++?

**Comment by raklaptudirm** [comment]
> @raklaptudirm, should I change the workflow to create a PR to the repository, just like in other repositories such as C/C++?

That would be best imo.

**Comment by Panquesito7** [comment]
If anyone wants to work on this, feel free to create a PR. 🙂

**Comment by SharmilaChandariah** [comment]
I can Check on this

**Comment by Sayan-ghs** [comment]
I would like to contribute on this
